src/plots/cartesian/axes.js

```diff
@@ -1136,9 +1147,26 @@ function formatDate(ax, out, hover, extraPrecision) {
             }
         }
     }
-    if(suffix && (!ax._inCalcTicks || (suffix !== ax._prevSuffix))) {
-        tt += suffix;
-        ax._prevSuffix = suffix;
+    if(ax.tickmode === 'array') {
+        // we get extra precision in array mode, but it may be useless, strip it off
+        if(tt === '00:00:00') {
+            tt = suffix;
+            suffix = '';
+        }
+        else {
+            tt = tt.replace(/:00$/, '');
+        }
+    }
+
+    if(suffix) {
+        if(hover) {
+            // hover puts it all on one line, so suffix works best up front
+            tt = suffix + (tt ? ' ' + tt : '');
```

Comment:
Hmm. 

Looks we're getting a little too much information on regular date axis (e.g. the `date_axis` mock)

![gifrecord_2016-11-22_170028](https://cloud.githubusercontent.com/assets/6675409/20543824/55cce3fe-b0d5-11e6-8b75-caf3952f0fe1.gif)